### PR TITLE
[libc++][TZDB] Renames incomplete.

### PR DESCRIPTION
**Pull request by mordante**
The new name uses experimental which better conveys what it means.

### CODE REVIEW

File: libcxx/include/__config
Comment (mordante) [inline libcxx/include/__config:422]: @philnik777 @ldionne should we rename PSTL too?

Comment (philnik777) [reply libcxx/include/__config:422]: I'll let Louis decide. He's currently working on some PSTL refactoring, so maybe not right now.

**Review by philnik777 — COMMENTED** [review]
#89178 is related I think. If we want to go ahead with that, maybe we want to rename this to `_LIBCPP_HAS_EXPERIMENTAL_TZDB` or `_LIBCPP_HAS_EXPERIMENTAL_FEATURE_TZDB`. FWIW just `_EXPERIMENTAL_` makes it sound to me like it's in `std::experimental`.

**Comment by mordante** [comment]
> #89178 is related I think. If we want to go ahead with that, maybe we want to rename this to `_LIBCPP_HAS_EXPERIMENTAL_TZDB` or `_LIBCPP_HAS_EXPERIMENTAL_FEATURE_TZDB`. FWIW just `_EXPERIMENTAL_` makes it sound to me like it's in `std::experimental`.

The new name matches existing names. So I prefer to have one pattern. This name was recently discussed on Discord too.

**Review by EricWF — APPROVED** [review]
Thank you.

**Comment by mordante** [comment]
> #89178 is related I think. If we want to go ahead with that, maybe we want to rename this to `_LIBCPP_HAS_EXPERIMENTAL_TZDB` or `_LIBCPP_HAS_EXPERIMENTAL_FEATURE_TZDB`. FWIW just `_EXPERIMENTAL_` makes it sound to me like it's in `std::experimental`.

Since at the moment #89178 does not change any of the experimental macros I'll land this now. We can change it afterwards; preferably when most of the new TZDB parts have landed. I've done one rename round after the Discord discussion and I'd prefer not to do a second one. (After everything has landed I still need to do quite a bit of work to get a better performance.)

**Pull request merged**
